pymeasure/display/widgets.py

```diff
@@ -490,16 +492,69 @@ def _layout(self):
         vbox = QtGui.QVBoxLayout(self)
         vbox.setSpacing(6)
 
+        self.labels = {}
         parameters = self._procedure.parameter_objects()
         for name in self._inputs:
             if not isinstance(getattr(self, name), self.NO_LABEL_INPUTS):
                 label = QtGui.QLabel(self)
                 label.setText("%s:" % parameters[name].name)
                 vbox.addWidget(label)
+                self.labels[name] = label
+
             vbox.addWidget(getattr(self, name))
 
         self.setLayout(vbox)
 
+    def _setup_visibility_groups(self):
+        groups = {}
+        parameters = self._procedure.parameter_objects()
+        for name in self._inputs:
+            parameter = parameters[name]
+            group = parameter.group_by
+            condition = parameter.group_condition
+            if group is None or group not in self._inputs or group == name:
+                continue
+
+            if isinstance(getattr(self, group), BooleanInput):
+                coondition = 2 if condition else 0
+
+            if group not in groups:
+                groups[group] = []
+
+            groups[group].append((getattr(self, name), condition))
+            if name in self.labels:
+                groups[group].append((self.labels[name], condition))
+
+        for name, group in groups.items():
+            def toggle(state):
+                for (el, con) in group:
+                    if callable(con):
+                        visible = con(state)
+                    else:
+                        visible = (state == con)
+
+                    if self._hide_groups:
+                        el.setHidden(not visible)
+                    else:
+                        el.setDisabled(not visible)
+
+            group_el = getattr(self, name)
+            if isinstance(group_el, BooleanInput):
+                group_el.stateChanged.connect(toggle)
+                toggle(group_el.checkState())
+            elif isinstance(group_el, StringInput):
+                group_el.textChanged.connect(toggle)
+                toggle(group_el.text())
+            elif isinstance(group_el, (IntegerInput, ScientificInput)):
+                group_el.valueChanged.connect(toggle)
+                toggle(group_el.value())
+            elif isinstance(group_el, ListInput):
+                group_el.currentTextChanged.connect(toggle)
+                toggle(group_el.currentText())
+            else:
+                raise NotImplementedError(
+                    "Grouping based on % (%) is not implemented." % (name, group_el))
+
```

Comment:
I don't know the format character `%` standalone. Did you mean `%s` ?

Reply:
Yes, you're right; I could've been a bit more careful with these last few commits